tests/test-url.lua

```diff
@@ -86,4 +86,24 @@ require('tap')(function(test)
       assert(actual == expected, 'relative test #' .. testNo .. ': ' .. actual .. ' should equal '.. expected)
     end)
   end
+
+  test('@ in path field', function()
+    local parsed = url.parse('/dir/@file')
+    assert(parsed.pathname == '/dir/@file')
+    assert(parsed.path == '/dir/@file')
+    assert(parsed.href == '/dir/@file')
+    assert(parsed.auth == nil)
+  end)
+
+  test('@ not in path field', function()
+    local parsed = url.parse('http://user:passwd@host:81/')
+    assert(parsed.pathname == '/')
+    assert(parsed.auth == 'user:passwd')
+    assert(parsed.path == '/')
+    assert(parsed.host == 'host:81')
+    assert(parsed.href == 'http://user:passwd@host:81/')
+    assert(parsed.port == '81')
+    assert(parsed.protocol == 'http')
+    assert(parsed.hostname == 'host')
+  end)
```

Comment:
These can be added to the `parseTests` table instead:

This can be added to the `parseTests` table above instead:

```
  ["/dir/@file"] = {href = '/dir/@file', path = '/dir/@file', pathname = '/dir/@file'},
  ["http://user:passwd@host:81/"] = {href = 'http://user:passwd@host:81/', protocol = 'http', auth = 'user:passwd', host = 'host:81', hostname = 'host', port = 81, path = '/', pathname = '/'},
```